File: binaries/cli/src/template/python/listener/listener-template.py
"""Module to process node input events and print received messages."""

from dora import Node

def main():
    """Listen for input events and print received messages."""
    node = Node()
    for event in node:
        if event["type"] == "INPUT":
            message = event["value"][0].as_py()
            print(f"""I heard {message} from {event["id"]}""")

if __name__ == "__main__":
    main()

File: binaries/cli/src/template/cxx/listener-template.cc
#include "dora-node-api.h" // adjust this path if necessary

#include <iostream>
#include <vector>

int main()
{
    std::cout << "HELLO FROM C++" << std::endl;
    unsigned char counter = 0;

    auto dora_node = init_dora_node();

    while (1)
    {
        auto event = dora_node.events->next();
        auto ty = event_type(event);

        if (ty == DoraEventType::AllInputsClosed)
        {
            break;
        }
        else if (ty == DoraEventType::Input)
        {
            auto input = event_as_input(std::move(event));
            auto input_id = input.id;
            auto message = std::string(reinterpret_cast<const char*>(input.data.data()), input.data.size());
            std::cout << "I heard " << message << " from " << std::string(input_id) << std::endl;
        } 
        else {
            std::cerr << "Unknown event type " << static_cast<int>(ty) << std::endl;
        }
    }

    return 0;;
}
